lib/ticker.dart

```diff
@@ -68,48 +68,38 @@ mixin Tick {}
 
 @immutable
 class ImmutableTick extends BlocState with Tick {
-  const ImmutableTick();
-  @override
-  int get hashCode => 0;
-  @override
-  bool operator ==(Object obj) => obj is Tick;
+  const ImmutableTick(TickType tickType) : super(tickType);
```

Comment:
removed `hashCode` and `==` overrides because you want different instances of `ImmutableTick` to be treated as non-equal.